src/Microsoft.Sbom.Parsers.Spdx22SbomParser/Utils/InternalMetadataProviderIdentityExtensions.cs

```diff
@@ -168,6 +168,27 @@ public static string GetDocumentNamespace(this IInternalMetadataProvider interna
         return internalMetadataProvider.GetSBOMNamespaceUri();
     }
 
+    public static string GetSwidTagId(this IInternalMetadataProvider internalMetadataProvider)
+    {
+        if (internalMetadataProvider is null)
+        {
+            throw new ArgumentNullException(nameof(internalMetadataProvider));
+        }
+
+        string rootPackageVersion = internalMetadataProvider.GetPackageVersion();
+        string packageSupplierFromMetadata = internalMetadataProvider.GetPackageSupplier();
+        string rootPackageName = internalMetadataProvider.GetPackageName();
+
+        Uri namespaceUri = new Uri(internalMetadataProvider.GetSBOMNamespaceUri());
+
+        // Generate a guid for the new swid tag Id.
+        string tagId = Guid.NewGuid().ToString();
+
+        Uri swidPurl = new Uri($"pkg:swid/{packageSupplierFromMetadata}/{namespaceUri.Host}/{rootPackageName}@{rootPackageVersion}?tag_id={tagId}");
```

Comment:
Would creating a Uri do the encoding that Aasim suggested in a previous iteration?

Reply:
Encoding would create a string like this "pkg%3Aswid%2FMicrosoft%2Fmicrosoft.com%2FMicrosoftSQLServer%403.2.12%3Ftag_id%3Dc5bffa01-30a9-45f5-b8a4-7dafd0f27441"

We don't do this for any of the purls in packages. Do we want to make this change?

Reply:
I think the suggestion was only encoding some parts such as  packageSupplierFromMetadata, rootPackageName androotPackageName